docs/internal.md

```diff
@@ -0,0 +1,80 @@
+- Updating the SAM translator dependency: https://github.com/aws-cloudformation/cfn-python-lint/pull/1536
+- Releasing a new version of cfn-lint: https://github.com/aws-cloudformation/cfn-python-lint/pull/1530
+- Releasing a new version of the companion VS Code plugin: https://github.com/aws-cloudformation/aws-cfn-lint-visual-studio-code/pull/76
+
+
+## Property data
+
+The precision of the linter depends on having up-to-date resource specifications that model the properties accurately. The rules use this property data for all the validations. 
+
+### Updating it
+
+The official resource specification is updated on a weekly basis (every Friday), so every week we update the property data by:
+
+```
+pip3 install -e .
+scripts/update_specs_from_pricing.py # requires Boto3 and Credentials
+scripts/update_specs_services_from_ssm.py # requires Boto3 and Credentials
+cfn-lint --update-specs
+cfn-lint --update-iam-policies
+cfn-lint --update-documentation
+```
+
+### Folder structure
+
+The official resource specifications are one source of data, the other two are the "extended specs" which are "patches" to the spec that enforce more constraints, and the "additional specs" which are rules written in JSON format that are then picked up by their respective Python class.
+
+#### CloudSpecs
+
+The command `cfn-lint --update-specs` pulls down the official resource specifications into folder `CloudSpecs` and patches the JSON files with the contents of the files in `ExtendedSpecs`. The merged results are stored in `CloudSpecs`.
+
+#### ExtendedSpecs
+
+- [`ExtendedSpecs/$REGION/05_pricing_property_values.json`](https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/data/ExtendedSpecs/all/05_pricing_property_values.json) written by [`scripts/update_specs_from_pricing.py`](https://github.com/aws-cloudformation/cfn-python-lint/blob/cc6ac28ff7deba86cb82813733cceec4bdff68a2/scripts/update_specs_from_pricing.py#L235)
+- [`ExtendedSpecs/$REGION/06_ssm_service_removal.json`](https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/data/ExtendedSpecs/us-gov-east-1/06_ssm_service_removal.json) written by [`scripts/update_specs_services_from_ssm.py`](https://github.com/aws-cloudformation/cfn-python-lint/blob/cc6ac28ff7deba86cb82813733cceec4bdff68a2/scripts/update_specs_services_from_ssm.py#L165)
+- [`ExtendedSpecs/$REGION/07_ssm_service_addition.json`](https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/data/ExtendedSpecs/us-gov-east-1/07_ssm_service_addition.json) written by [`scripts/update_specs_services_from_ssm.py`](https://github.com/aws-cloudformation/cfn-python-lint/blob/cc6ac28ff7deba86cb82813733cceec4bdff68a2/scripts/update_specs_services_from_ssm.py#L204)
+- Allowed patterns:
+https://github.com/aws-cloudformation/cfn-python-lint/blob/a46773c752247c51effef415bd3462eaec10ab0b/src/cfnlint/data/ExtendedSpecs/all/03_value_types.json#L33-L36
```

Comment:
[reason I'd left this documentation in a Github comment is that the code previews render there, whereas they don't render in markdown source code](https://github.com/aws-cloudformation/cfn-python-lint/issues/903#issuecomment-595356853)

Reply:
I inlined the examples in the latest commit